daemon/cluster/cluster.go

```diff
@@ -1070,16 +1070,21 @@ func (c *Cluster) ServiceLogs(ctx context.Context, input string, config *backend
 		return c.errNoManager(state)
 	}
 
+	var selector swarmapi.LogSelector
 	service, err := getService(ctx, state.controlClient, input)
```

Comment:
`getService` could potentially return a different error than "service not found", such as an ambiguous service ID prefix or a problem communicating with the API. I think it's a good idea to check for the "service not found" error specifically by checking `GetHTTPErrorStatusCode(err) == http.StatusNotFound`. If some other error is returned, we should return that error instead of trying `getTask`.